WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -86,7 +87,38 @@ final class TotalsViewModel: ObservableObject {
         startSyncingOrder(with: cartItems, allItems: allItems)
     }
 
+    private func areOrderAndCartDifferent(cartItems: [CartItem]) -> Bool {
+        // check if order has same items as cart does
+        if let order {
+            // first we get list of all products 1 by 1
+            var cleanOrderItems: [POSOrderItem] = order.items.flatMap {
+                Array(repeating: $0, count: $0.quantity.intValue)
+            }
+            var cleanCartItems: [POSItem] = cartItems.flatMap {
+                Array(repeating: $0.item, count: $0.quantity)
+            }
+            if cleanOrderItems.count == cleanCartItems.count {
```

Comment:
Question: Is it enough to compare if the number of items are the same in order to continue? Or we need to perform some more validation?

Reply:
The order items and cart items need to be exactly the same. If their count is different then we are sure that they are different and we do not need to do any more checks. If the count is the same then we check if they have the same items.
I added this check to avoid doing detailed check if it is not needed.